### PR TITLE
Add recipes for mammoth & markitdown

### PR DESCRIPTION
**Pull request by johnhany97**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by johnhany97** [comment]
I confirm that i'm willing to be listed as a maintainer of these 3 packages.

File: recipes/cobble/LICENSE [hunk 1]
```diff
@@ -0,0 +1,12 @@
+BSD-2-Clause
```
Comment (johnhany97) [inline recipes/cobble/LICENSE:1]: `cobble` doesn't currently commit the LICENSE file. They however reference it in the README of that repo [here](https://github.com/mwilliamson/python-cobble/tree/a4f92a3?tab=readme-ov-file#license).

File: recipes/mammoth/fix_build_on_windows.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff --git a/setup.py b/setup.py
+index 0e4c48f..f3c6ce3 100644
+--- a/setup.py
++++ b/setup.py
+@@ -4,7 +4,7 @@ import os
+ from setuptools import setup
+
+ def read(fname):
+-    return open(os.path.join(os.path.dirname(__file__), fname)).read()
++    return open(os.path.join(os.path.dirname(__file__), fname), encoding="utf8").read()
```
Comment (johnhany97) [inline recipes/mammoth/fix_build_on_windows.patch:10]: Not doing so yielded errors when building this library on windows.
```
  Traceback (most recent call last):
    File "D:\bld\mammoth_1734340231977\_h_env\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
      main()
    File "D:\bld\mammoth_1734340231977\_h_env\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "D:\bld\mammoth_1734340231977\_h_env\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 149, in prepare_metadata_for_build_wheel
      return hook(metadata_directory, config_settings)
    File "D:\bld\mammoth_1734340231977\_h_env\lib\site-packages\setuptools\build_meta.py", line 366, in prepare_metadata_for_build_wheel
      self.run_setup()
    File "D:\bld\mammoth_1734340231977\_h_env\lib\site-packages\setuptools\build_meta.py", line 311, in run_setup
      exec(code, locals())
    File "<string>", line 14, in <module>
    File "<string>", line 7, in read
    File "D:\bld\mammoth_1734340231977\_h_env\lib\encodings\cp1252.py", line 23, in decode
      return codecs.charmap_decode(input,self.errors,decoding_table)[0]
  UnicodeDecodeError: 'charmap' codec can't decode byte 0x9d in position 3430: character maps to <undefined>
```
[relevant CI job](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1111141&view=logs&j=53c3d5f4-e7c7-5b53-4b94-211701b2ba17&t=8f042650-8479-5761-7a4a-8d021c0bb2de)

File: recipes/markitdown/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
+{% set name = "markitdown" %}
+{% set version = "0.0.1a2" %}
```
Comment (ocefpaf) [inline recipes/markitdown/meta.yaml:2]: This one falls under https://github.com/conda-forge/cfep/blob/main/cfep-05.md 

```suggestion
{% set version = "0.0.1a2" %}
```

File: recipes/markitdown/meta.yaml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {% set name = "markitdown" %}
-{% set version = "0.0.1a2" %}
+{% set version = "0.0.1dev2" %}
```
Comment (ocefpaf) [inline recipes/markitdown/meta.yaml]: You have to follow CFEP-05 or ask upstream for a stable release.

Comment (johnhany97) [reply recipes/markitdown/meta.yaml]: I changed this back to `0.0.1a2`, but now override the channel target - would that be an acceptable path forwards?

Comment (ocefpaf) [reply recipes/markitdown/meta.yaml]: You only need to remember to remove the conda `conda_build_config.yaml` file when updating this to a stable release.

Comment (johnhany97) [reply recipes/markitdown/meta.yaml]: Makes sense 👍

File: recipes/mammoth/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
+{% set name = "mammoth" %}
```
Comment (ocefpaf) [inline recipes/mammoth/meta.yaml:1]: I believe this is the same as https://github.com/conda-forge/staged-recipes/pull/28583

@sarthakpati  do you want to be added as a maintainer here so we can merge this one?

Comment (sarthakpati) [reply recipes/mammoth/meta.yaml:1]: Sure thing, thanks!

Comment (johnhany97) [reply recipes/mammoth/meta.yaml:1]: done in 7222a1518623f30af21df47ed2f8d6abeffe23cc ✅

File: recipes/markitdown/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+about:
+  home: https://github.com/microsoft/markitdown
+  summary: Python tool for converting files and office documents to Markdown.
+  dev_url: https://github.com/microsoft/markitdown
+  license: MIT
+  license_file: LICENSE
+
+extra:
+  recipe-maintainers:
+    - johnhany97
```
Comment (johnhany97) [inline recipes/markitdown/meta.yaml:61]: @sarthakpati do you also want to be listed as a recipe maintainer for `markitdown`?

Comment (sarthakpati) [reply recipes/markitdown/meta.yaml:61]: Sure, thanks!

Comment (ocefpaf) [reply recipes/markitdown/meta.yaml:61]: Please do that in the feedstock. Thank you both!

**Pull request merged**